docs/migrations/rest.md

```diff
@@ -1,101 +0,0 @@
-<!--
-order: 2
--->
-
-# REST Endpoints Migration
```

Comment:
should we actually keep this doc for 0.44? They are removed in 0.44, so a user who learns about this removal during 0.44 might want to refer to this doc

Reply:
I added it back with revisions. Let me know if the revisions make sense.

Reply:
Looks good!